components/twitch/sources/common-webhook.js

```diff
@@ -0,0 +1,77 @@
+const common = require("./common.js");
+const subscriptionExpiration = 864000; // seconds until webhook subscription expires, maximum 10 days (864000 seconds)
+
+module.exports = {
+  ...common,
+  props: {
+    ...common.props,
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: subscriptionExpiration,
+      },
+    },
+  },
+  hooks: {
+    async activate() {
+      const topics = await this.getTopics();
+      this.db.set("topics", topics);
+      await this.subscribeToHooks(topics);
+    },
+    async deactivate() {
+      const topics = this.db.get("topics");
+      for (const topic of topics)
+        await this.twitch.manageHook(
+          "unsubscribe",
+          this.http.endpoint,
+          topic,
+          subscriptionExpiration
+        );
+    },
+  },
+  methods: {
+    ...common.methods,
+    async subscribeToHooks(topics) {
```

Comment:
Can we change this a bit so that we can use it in `deactivate()` as well?
```js
module.exports = {
  hooks: {
    async activate() {
      const topics = await this.getTopics();
      this.db.set("topics", topics);
      await this.manageHookForTopics("subscribe", topics);
    },
    async deactivate() {
      const topics = this.db.get("topics");
      await this.manageHookForTopics("unsubscribe", topics);
    },
  },
  methods: {
    manageHookForTopics(mode, topics) {
      for (const topic of topics) {
        await this.twitch.manageHook(
          mode,
          this.http.endpoint,
          topic,
          subscriptionExpiration,
        );
      }
    },
  },
};
```

We should also add a `try/catch` block around the `manageHook` call so that we don't leave active subscriptions in case one of the calls fails.

Reply:
Updated to include a try/catch block in manageHookForTopics method.